WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsViewController.swift

```diff
@@ -0,0 +1,177 @@
+
+    init(product: Product) {
+        viewModel = ProductSettingsTableViewModel(product: product)
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureNavigationBar()
+        configureTableView()
+    }
+
+}
+
+// MARK: - View Configuration
+//
+private extension ProductSettingsViewController {
+
+    func configureNavigationBar() {
+        title = NSLocalizedString("Product Settings", comment: "Product Settings navigation title")
+
+        removeNavigationBackBarButtonText()
+    }
+
+    func configureMainView() {
+        view.backgroundColor = .listBackground
+    }
+
+    func configureTableView() {
+        viewModel.registerTableViewCells(tableView)
+        viewModel.registerTableViewHeaderFooters(tableView)
+
+        tableView.dataSource = self
+        tableView.delegate = self
+
+        tableView.backgroundColor = .listBackground
+        tableView.removeLastCellSeparator()
+
+        tableView.reloadData()
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension ProductSettingsViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return viewModel.sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let section = viewModel.sections[section]
+        switch section {
+        case .publishSettings( _, let rows):
+            return rows.count
+        case .moreOptions( _, let rows):
+            return rows.count
+        }
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let section = viewModel.sections[indexPath.section]
+        let reuseIdentifier = section.reuseIdentifier(at: indexPath.row)
+        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier, for: indexPath)
+        configure(cell, section: section, indexPath: indexPath)
+        return cell
+    }
+
+    func configure(_ cell: UITableViewCell, section: ProductSettingsSection, indexPath: IndexPath) {
+        switch section {
+        case .publishSettings( _, let rows):
+            configureCellInPublishSettingsSection(cell, row: rows[indexPath.row])
+        case .moreOptions( _, let rows):
+            configureCellInMoreOptionsSection(cell, row: rows[indexPath.row])
+        }
+    }
+}
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension ProductSettingsViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
```

Comment:
We always try to configure all the things from code, just because it's easily debuggable when we work in a team. 
Eg: check for example this issue #1034 a lot of problems because the tableview isn't responding to taps, but the solution was simple #1856: the Tap had also been disabled in xib file.

Reply:
> We always try to configure all the things from code, just because it's easily debuggable when we work in a team.

I can empathize on this. And thanks for the context.

My concern is that for this specific case, we're inconsistent because the estimated row height is declared in the xib.

![image](https://user-images.githubusercontent.com/198826/74461587-f9e0e280-4e4b-11ea-94c4-8342426d72dd.png)

It can be confusing which one we should we really do. I wonder, since we prefer code over xib, it might be better to fully declare the `UITableView` as code. Only for cases like this where the xib really just contains a `UITableView`. If it's a cell with a complicated layout, it's better if it's in a xib. 

---

In other news, it looks like the [default of `estimatedSectionHeaderHeight`](https://developer.apple.com/documentation/uikit/uitableview/1614957-estimatedsectionheaderheight) is `.automaticDimension` so perhaps we don't need this. 

Oh, wait, the xib needs to be fixed. 🤦‍♂